velox/expression/VectorFunction.cpp

```diff
@@ -59,6 +60,12 @@ std::shared_ptr<VectorFunction> getVectorFunction(
     });
   }
 
+  if (resultType) {
+    // For certain vector functions that involve Decimal types, the return type
```

Comment:
This sounds strange. Input types should be sufficient to determine the result type, no?

Reply:
Yes, we can infer form input args, but we need to recompute the return type Precision and Scale which is already done at the time of binding. So passing result type will help us reuse the already computed return type.

Reply:
This calculation is rather cheap. Is it necessary to reuse?

Reply:
@mbasmanova For decimal arithmetic, the decimal return type is computed using expression/type calculation. Like @karteekmurthys  said, our goal here is to avoid the binding/re-computation or return type from the input arguments.

Reply:
I'm a bit confused. Factory function for creating vector function receives actual types, which can be queried for precision and scale. From these 2 numbers, the rescale factors can be computed. No need to use expressions. Am I missing something?

Reply:
Changing the signature of the factory function is problematic, especially, given variadic arguments.

Reply:
You are right. We are seeing test failures.
The decmalAddSignature is as follows
```
exec::FunctionSignatureBuilder()
          .returnType("DECIMAL(r_precision, r_scale)")
          .argumentType("DECIMAL(a_precision, a_scale)")
          .argumentType("DECIMAL(b_precision, b_scale)")
          .variableConstraint(
              "r_precision",
              "min(38, max(a_precision - a_scale, b_precision - b_scale) + max(a_scale, b_scale) + 1)")
          .variableConstraint("r_scale", "max(a_scale, b_scale)")
          .build()};
```
Our goal was to avoid recomputing and to keep the API clean. Since we cannot pass return type, there are two options inside the Decimal Function Factory.
Create a SignatureBinder using the arguments and signature specified above, and invoke tryResolveReturnType().
or
Create another function by duplicating the expression logic "min(38, max(a_precision - a_scale, b_precision - b_scale) + max(a_scale, b_scale) + 1)" and infer the return type.

Reply:
The issue is with the return type and not the rescale factors. The return type is required to create the vector function since the return value could either be a ShortDecimal or a LongDecimal based on the input arguments.

Reply:
> Create another function by duplicating the expression logic "min(38, max(a_precision - a_scale, b_precision - b_scale) + max(a_scale, b_scale) + 1)" and infer the return type.

This should be fine.